packages/core/src/Stores/client-store.js

```diff
@@ -763,95 +766,89 @@ export default class ClientStore extends BaseStore {
-            const is_maltainvest_account = this.root_store.ui.real_account_signup_target === 'maltainvest';
-            let currency = '';
-            form_values.residence = this.residence;
+    async realAccountSignup(form_values) {
+        const is_maltainvest_account = this.root_store.ui.real_account_signup_target === 'maltainvest';
+        let currency = '';
+        form_values.residence = this.residence;
+        if (is_maltainvest_account) {
+            currency = form_values.currency;
+            form_values.accept_risk = 1;
+            delete form_values.currency;
+        }
+        const response = is_maltainvest_account
+            ? await WS.newAccountRealMaltaInvest(form_values)
+            : await WS.newAccountReal(form_values);
+        if (!response.error) {
+            await this.accountRealReaction(response);
+            // Set currency after account is created
+            // Maltainvest only
             if (is_maltainvest_account) {
-                currency = form_values.currency;
-                form_values.accept_risk = 1;
-                delete form_values.currency;
+                await this.setAccountCurrency(currency);
             }
-            const response = is_maltainvest_account
-                ? await WS.newAccountRealMaltaInvest(form_values)
-                : await WS.newAccountReal(form_values);
-            if (!response.error) {
-                await this.accountRealReaction(response);
-                // Set currency after account is created
-                // Maltainvest only
-                if (is_maltainvest_account) {
-                    await this.setAccountCurrency(currency);
-                }
-                localStorage.removeItem('real_account_signup_wizard');
-                this.root_store.gtm.pushDataLayer({ event: 'real_signup' });
-                resolve({
-                    ...response,
-                    ...(is_maltainvest_account
-                        ? {
-                              new_account_maltainvest: {
-                                  ...response.new_account_maltainvest,
-                                  currency,
-                              },
-                          }
-                        : {}),
-                });
-            } else {
-                reject(response.error);
-            }
-        });
+            localStorage.removeItem('real_account_signup_wizard');
+            this.root_store.gtm.pushDataLayer({ event: 'real_signup' });
+            Promise.resolve({
+                ...response,
+                ...(is_maltainvest_account
+                    ? {
+                          new_account_maltainvest: {
+                              ...response.new_account_maltainvest,
+                              currency,
+                          },
+                      }
+                    : {}),
+            });
+        } else {
+            Promise.reject(response.error);
```

Comment:
Should we `throw` here instead? This seems to just create a rejected `Promise` which isn't returned.